app/models/miq_action.rb

```diff
@@ -829,6 +829,26 @@ def action_ems_refresh(action, rec, inputs)
     end
   end
 
+  def action_container_image_analyze(action, rec, inputs)
+    action_method = "action_container_image_analyze"
+    # This event does not yes exists but better be on the safe side
+    if inputs[:event].name == "request_container_image_scan"
+      MiqPolicy.logger.warn("MIQ(#{action_method}): Invoking action [#{action.description}] for event"\
+                            " [#{inputs[:event].description}] would cause infinite loop, skipping")
+      return
+    end
+
+    unless rec.kind_of?(ContainerImage)
+      MiqPolicy.logger.error("MIQ(#{action_method}): Unable to perform action [#{action.description}],"\
+                             " object [#{rec.inspect}] is not a Container Image")
+      return
```

Comment:
Can you put this guard clause first?  It seems like a more general problem we want to avoid, whereas the first guard clause (infinite loop) is a more specific condition we want to guard against.